membership-attribute-service/app/services/AttributesFromZuora.scala

```diff
@@ -63,14 +62,14 @@ object AttributesFromZuora extends LoggingWithLogstashFields {
 
       if(fromWhere == "Zuora") {
         attributesFromDynamo map { case (_, dynamoAttributes) =>
-          val canSkipCacheUpdate: Boolean = dynamoAndZuoraAgree(dynamoAttributes, attributes, identityId)
           val zuoraAttributesWithAdfree: Option[Attributes] = attributesWithAdFreeFlagFromDynamo(attributes, dynamoAttributes)
 
-          if(!canSkipCacheUpdate) {
+          val updateRequired = dynamoUpdateRequired(dynamoAttributes, attributes, identityId)
+          if(updateRequired) {
             updateCache(identityId, zuoraAttributesWithAdfree, dynamoAttributeService).onFailure {
```

Comment:
updateCache returns a Future, but I can't see where we use that future.  This shouldn't really be left hanging even if scala's dubious concurrency model lets us.